lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/python
+# coding: utf-8
+#
+# Copyright: (c) 2019, Vladimir Porshkevich (@porshkevich) <neosonic@mail.ru>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = u'''
+module: docker_plugin
+version_added: "2.8"
+short_description: Manage Docker plugins
+description:
+  - Install/remove Docker plugins.
+  - Performs largely the same function as the "docker plugin" CLI subcommand.
+options:
+  name:
+    description:
+      - Name of the plugin to operate on.
+    required: true
+    type: str
+
+  alias:
+    description:
+      - Alias of the plugin to operate on. Same plugin can be installed with different alias.
+    type: str
+
+  plugin_options:
+    description:
+      - Dictionary of plugin settings.
+    type: dict
+
+  state:
+    description:
+      - C(absent) remove the plugin.
+      - C(present) install the plugin, if it does not already exist.
+      - C(enable) enable the plugin.
+      - C(disable) disable the plugin.
```

Comment:
One question is whether `enable` includes `present` or not. There are two philosophies: either `enabled` and `disabled` assume that the plugin is installed, or they install the plugin if it isn't installed. You're currently following the first one. I don't care what you want to do in the end, I just want to mention this in case you didn't knew/thought about the second.

Reply:
Is the `upgrade` feature not implemented intentionally?

Reply:
I'm not sure whether upgrade should be another state, or a flag (in the latter case, it's orthogonal to `present`/`enable`/`disable`).

Reply:
In either way it's not covered

Reply:
Initially, I did not set myself the task of covering the possibility of `upgrade`. as @felixfontein said, i'm not sure if it can be done as a separate state. I am ready to listen to the proposals, but this should not interfere with the adoption of this pool request

Reply:
You should probably mention in the module's description that it currently doesn't support upgrading plugins.

Reply:
If set `alias` option as required, we can use `upgrade` as state. By alias can find the plugin and call disable and upgrade methods. But this work only with plugins with `alias`.

Reply:
You can make `alias` a `required_if` in case `state` equals `upgrade`. See for example https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py#L2928-L2934 where `image` is required if `state` equals `present`.